### ISSUE TITLE
FlutterLogEntryParser exceptions

### ISSUE THREAD

**Issue by darrenaustin** [flutter/flutter-intellij#3693]
## What happened

Just using the plugin to write and run flutter apps, I have been seeing this
error quite often. The only possibly weird thing in my setup is that I am working
on the Flutter framework, so my Flutter SDK path is pointing at code I am editing.

## Version information

IntelliJ IDEA `2019.1.3` • Flutter plugin `io.flutter 37.1.3` • Dart plugin `191.7830`

Flutter 1.7.5-pre.361 • channel master • git@github.com:flutter/flutter.git
Framework • revision 819538704a (22 hours ago) • 2019-07-22 17:08:42 -0700
Engine • revision 9e9da56b89
Tools • Dart 2.5.0 (build 2.5.0-dev.1.0 a089199b93)

## Exception

null

```
java.lang.reflect.InvocationTargetException
	at sun.reflect.GeneratedMethodAccessor157.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.execution.process.ProcessHandler$2.invoke(ProcessHandler.java:214)
	at com.sun.proxy.$Proxy23.onTextAvailable(Unknown Source)
	at com.intellij.execution.process.ProcessHandler.notifyTextAvailable(ProcessHandler.java:188)
	at com.intellij.execution.process.BaseOSProcessHandler$SimpleOutputReader.onTextAvailable(BaseOSProcessHandler.java:179)
	at com.intellij.util.io.BaseOutputReader.sendText(BaseOutputReader.java:211)
	at com.intellij.util.io.BaseOutputReader.processInput(BaseOutputReader.java:195)
	at com.intellij.util.io.BaseOutputReader.readAvailableBlocking(BaseOutputReader.java:149)
	at com.intellij.util.io.BaseDataReader.readAvailable(BaseDataReader.java:77)
	at com.intellij.util.io.BaseDataReader.doRun(BaseDataReader.java:155)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:224)
	at com.intellij.util.io.BaseDataReader.lambda$start$0(BaseDataReader.java:61)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	at java.lang.String.substring(String.java:1960)
	at io.flutter.logging.text.LineParser.parse(LineParser.java:112)
	at io.flutter.logging.FlutterLogEntryParser$LineHandler.parseLineStyle(FlutterLogEntryParser.java:289)
	at io.flutter.logging.FlutterLogEntryParser$LineHandler.parseLineInfo(FlutterLogEntryParser.java:307)
	at io.flutter.logging.FlutterLogEntryParser$LineHandler.parseEntry(FlutterLogEntryParser.java:312)
	at io.flutter.logging.FlutterLogEntryParser.parseDaemonEvent(FlutterLogEntryParser.java:348)
	at io.flutter.logging.FlutterLogEntryParser.parseDaemonEvent(FlutterLogEntryParser.java:272)
	at io.flutter.logging.FlutterLog$1.onTextAvailable(FlutterLog.java:168)
	... 19 more
```

**Comment by pq** [comment]
Hi @darrenaustin.  It looks like you've got the experimental logging view enabled.  As a quick fix you might go back to the standard console for logging output (in your Flutter preferences).  We're working on a bunch of improvements in that area as well.

Thanks for the report and sorry for the inconvenience!

**Comment by darrenaustin** [comment]
That was it. I had totally forgotten that I had turned that on. It is a nice feature, so I look forward to turning it back on after your improvements come in. Thx.

**Comment by pq** [comment]
@stevemessick: this should likely be fixed by https://github.com/flutter/flutter-intellij/pull/3711?

**Comment by stevemessick** [comment]
This is fixed with #3711
